### PR TITLE
 persistent terraform token provider changes

### PR DESCRIPTION
**Pull request by ansalamdaniel**
### Description
This PR addresses the changes related to Terraform provider  to support persistent terraform token. It accepts the refresh token using that token it get the ID token for further processing.

### Issue

[NGPC-697](https://platform9.atlassian.net/browse/NGPC-697)

### CODE REVIEW

**Comment by ansalamdaniel** [comment]
@TheNilesh  Please review this PR

File: internal/provider/provider.go [hunk 1]
```diff
@@ -29,7 +32,8 @@ func New(version string) func() provider.Provider {
 }
 
 type spotProvider struct {
-	Version string
+	NgpcHTTPClient *ngpc.HTTPClient
```
Comment (TheNilesh) [inline internal/provider/provider.go]: Unused variable `NgpcHTTPClient`. Please remove it

File: internal/provider/provider.go [hunk 2]
```diff
@@ -57,31 +61,89 @@ func (p *spotProvider) Configure(ctx context.Context, req provider.ConfigureRequ
 	if resp.Diagnostics.HasError() {
 		return
 	}
+
+	NgpcCfg := ngpc.NewConfig(ngpcAPIServer, "", p.Version == "dev")
```
Comment (TheNilesh) [inline internal/provider/provider.go:66]: Add a comment explaination

File: internal/provider/provider.go (hunk 1 above)
Comment (ansalamdaniel) [reply internal/provider/provider.go]: Done.

File: internal/provider/provider.go (hunk 2 above)
Comment (ansalamdaniel) [reply internal/provider/provider.go:66]: Added comments.

**Review by TheNilesh — APPROVED** [review]

**Pull request merged**